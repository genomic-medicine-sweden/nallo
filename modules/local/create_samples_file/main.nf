process CREATE_SAMPLES_FILE {
    tag "${meta.id}"
    label 'process_single'

    conda "anaconda::gawk=5.1.0"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/gawk:5.1.0' :
        'biocontainers/gawk:5.1.0' }"

    input:
    tuple val(meta), path(txt)

    output:
    tuple val(meta), path("*.txt"), emit: samples
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "${meta.id}"
    if ("$txt" == "${prefix}.txt") error "Input and output names are the same, set prefix in module configuration to disambiguate!"

    """
    awk '{print \$1,"${meta.id}"}' ${txt} > ${prefix}.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        gawk: \$(awk -Wversion | sed '1!d; s/.*Awk //; s/,.*//')
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    touch ${prefix}.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        gawk: \$(awk -Wversion | sed '1!d; s/.*Awk //; s/,.*//')
    END_VERSIONS
    """
}
