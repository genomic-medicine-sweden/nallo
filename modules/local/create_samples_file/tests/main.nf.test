nextflow_process {

    name "Test Process CREATE_SAMPLES_FILE"
    script "../main.nf"
    process "CREATE_SAMPLES_FILE"

    tag "modules"
    tag "modules_local"
    tag "create_samples_file"

    config "./nextflow.config"

    test("create_samples_file") {

        when {
            process {
                """
                input[0] = Channel.of('test_vcf_file_sample')
                    .collectFile(name: 'test.txt')
                    .map { file -> [ [ id: 'test' ], file ] }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("create_samples_file -stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = Channel.of('test_vcf_file_sample')
                    .collectFile(name: 'test.txt')
                    .map { file -> [ [ id: 'test' ], file ] }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }
}
