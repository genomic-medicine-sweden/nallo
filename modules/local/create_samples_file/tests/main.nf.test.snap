{
    "create_samples_file -stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "samples_file.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,ef72b0fd7e245e40b0ce6d409e218689"
                ],
                "samples": [
                    [
                        {
                            "id": "test"
                        },
                        "samples_file.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ef72b0fd7e245e40b0ce6d409e218689"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-09-11T18:56:06.026507892"
    },
    "create_samples_file": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "samples_file.txt:md5,41e40eee31c3003c213685c180f52c25"
                    ]
                ],
                "1": [
                    "versions.yml:md5,ef72b0fd7e245e40b0ce6d409e218689"
                ],
                "samples": [
                    [
                        {
                            "id": "test"
                        },
                        "samples_file.txt:md5,41e40eee31c3003c213685c180f52c25"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ef72b0fd7e245e40b0ce6d409e218689"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-09-11T18:55:18.906428667"
    }
}